docs/resources/apig_certificate.md

```diff
@@ -0,0 +1,150 @@
+}
+```
+
+### Manages a local SSL certificate in a specified dedicated APIG instance
+
+```hcl
+variable "certificate_name" {}
+variable "certificate_content" {
+  type    = string
+  default = "'-----BEGIN CERTIFICATE-----THIS IS YOUR CERT CONTENT-----END CERTIFICATE-----'"
+}
+variable "certificate_private_key" {
+  type    = string
+  default = "'-----BEGIN PRIVATE KEY-----THIS IS YOUR PRIVATE KEY-----END PRIVATE KEY-----'"
+}
+variable "dedicated_instance_id" {}
+
+resource "huaweicloud_apig_application" "test" {
+  name        = var.certificate_name
+  content     = var.certificate_content
+  private_key = var.certificate_private_key
+  type        = "instance"
+  instance_id = var.dedicated_instance_id
+}
+```
+
+### Manages a local SSL certificate (with the ROOT CA certificate)
+
+```hcl
+variable "certificate_name" {}
+variable "certificate_content" {
+  type    = string
+  default = "'-----BEGIN CERTIFICATE-----THIS IS YOUR CERT CONTENT-----END CERTIFICATE-----'"
+}
+variable "certificate_private_key" {
+  type    = string
+  default = "'-----BEGIN PRIVATE KEY-----THIS IS YOUR PRIVATE KEY-----END PRIVATE KEY-----'"
+}
+variable "root_ca_certificate_content" {
+  type    = string
+  default = "'-----BEGIN CERTIFICATE-----THIS IS YOUR CERT CONTENT-----END CERTIFICATE-----'"
+}
+variable "dedicated_instance_id" {}
+
+resource "huaweicloud_apig_application" "test" {
+  name            = var.certificate_name
+  content         = var.certificate_content
+  private_key     = var.certificate_private_key
+  trusted_root_ca = var.root_ca_certificate_content
+  type            = "instance"
+  instance_id     = var.dedicated_instance_id
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region where the certificate is located.  
+  If omitted, the provider-level region will be used. Changing this will create a new resource.
+
+* `name` - (Required, String) Specifies the certificate name.  
+  The valid length is limited from `4` to `50`, only Chinese and English letters, digits and underscores (_) are
+  allowed. The name must start with a Chinese or English letter.  
+
+* `content` - (Required, String) Specifies the certificate content.  
+  Read these documentations to learn [how to purchase a private ROOT CA certificate](https://support.huaweicloud.com/intl/en-us/tg-ccm/ccm_01_0016.html)
+  and [how to apply a private SSL certificate](https://support.huaweicloud.com/intl/en-us/tg-ccm/ccm_01_0025.html).
+
+* `private_key` - (Required, String) Specifies the private key of the certificate.
+
+* `type` - (Optional, String, ForceNew) Specifies the certificate type. The valid values are as follows:
+  + **instance**
+  + **global**
+
+  Defaults to **global**. Changing this will create a new resource.
+
+* `instance_id` - (Optional, String, ForceNew) Specifies the dedicated instance ID to which the certificate belongs.  
+  Changing this will create a new resource.
```

Comment:
How about add :
This field is required when `type` is set to **instance**.

Reply:
okey